addon/lib/field.js

```diff
@@ -330,6 +331,33 @@ export default Base.extend({
     return this.question.isMultipleChoice ? selected : selected[0];
   }),
 
+  /**
+   * The field's JEXL context.
+   *
+   * Properties:
+   * - `form`: Legacy property pointing to the root form.
+   * - `info.form`: The form this question is attached to.
+   * - `info.parent.form`: The parent form if applicable.
+   * - `info.root.form`: The new property for the root form.
+   */
+  jexlContext: computed(
+    "document.jexlContext.{form,info.root.form}",
+    "fieldset.{form.slug,field.fieldset.form.slug}",
+    function() {
+      const context = cloneDeep(this.document.jexlContext);
+      context.info.form = this.get("fieldset.form.slug");
+
+      // Only attach the parent context if in a subform.
```

Comment:
Is this handled the same way in the backend or is the property just `None` if there is no parent?

Reply:
From how I interpreted what I've read so far I thought that parent is missing completely if not in a subform. But I will check.

Reply:
You were right! `parent` is implemented as a getter and thus always present.

https://github.com/projectcaluma/caluma/blob/master/caluma/caluma_form/structure.py#L28
Thank you and to @winged !

Reply:
@anehx  JEXL (at least pyjexl) evaluates non-existing identifiers to `None` and crashes if you do anything "incompatible" with them (like continuing the dot notation, or adding transforms...). I'd add a `parent` here and just set it to `null` or `undefined` for consistency

Reply:
> @anehx JEXL (at least pyjexl) evaluates non-existing identifiers to `None` and crashes if you do anything "incompatible" with them (like continuing the dot notation, or adding transforms...). I'd add a `parent` here and just set it to `null` or `undefined` for consistency

@fkm Alright, let's do that then

Reply:
Done